Framework/Subset_of_CorLib/System/Single.cs

```diff
@@ -22,14 +22,31 @@ public struct Single
 
         public override String ToString()
         {
-            return Number.Format(m_value, false, "G", NumberFormatInfo.CurrentInfo);
+            string str = ((Double)m_value).ToString();

```

Comment:
AFAIK, only Double and Single are using Number.Format method with these parameters. Double class has its own mechanism to find out special values and treat them accordingly. Double also does it at the managed layer (with the help of a few native functions). For the Single type, we do not have those native methods and implementing them is not trivial, so I got help from the Double type.
I agree the it would be better if you could fix it at the native layer but it requires a significant amount to effort (implementing a bunch of native method to recognize special values for the Single type) 
Similarly, bringing the logic to Number class requires modifications to Double types while does not add a lot of value.
I believe it is better to keep the change here for now.